### ISSUE TITLE
Tailwind CSS styles are broken on not-found page

### ISSUE THREAD

**Issue by Josehower** [vercel/next.js#47970]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: Ubuntu 20.04.0 LTS Wed Apr 05 2023 17:42:03 GMT+0200 (Central European Summer Time)
    Binaries:
      Node: 16.14.2
      npm: 7.17.0
      Yarn: 1.22.19
      pnpm: 7.13.6
    Relevant packages:
      next: 13.2.5-canary.30
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Link to the code that reproduces this issue

https://stackblitz.com/edit/vercel-next-js-brvnkq?file=tailwind.config.js,app%2Fglobal.css,app%2Fpage.tsx,app%2Flayout.tsx,app%2Fnot-found.tsx

### To Reproduce

- open the reproduction example
- click the link or - visit a page that triggers the not-found component
- verify the code and see that Tailwind CSS is broken

Home Page
![image](https://user-images.githubusercontent.com/61600906/230134755-d26d8c16-4165-47bd-bcbd-1ce609f4f8d6.png)
Not Found Page
![image](https://user-images.githubusercontent.com/61600906/230134742-b1023800-a655-4ccb-aed1-d619ab75ba09.png)

### Describe the Bug

since [13.2.5-canary.22](https://www.npmjs.com/package/next/v/13.2.5-canary.22) Tailwind CSS stop working in not-found page. It is hard to really say what the reason is, and I haven't found any workaround until now.

[example repo for canary-21 that is working as expected](https://stackblitz.com/edit/vercel-next-js-rppczb?file=tailwind.config.js,app%2Fglobal.css,app%2Fpage.tsx,app%2Flayout.tsx,package.json)

### Expected Behavior

Tailwind CSS should work on the not-found page

**Comment by karlhorky** [comment]
cc @shuding

**Comment by shuding** [comment]
Confirmed that it's caused by https://github.com/vercel/next.js/pull/47688. Will see if we can revert or get it fixed.

**Comment by huozhi** [comment]
Fixed in [v13.2.5-canary.33](https://github.com/vercel/next.js/releases/tag/v13.2.5-canary.33)

**Comment by karlhorky** [comment]
Thanks for this @huozhi ! 🙌 Just tested `next@13.2.5-canary.33` in our application and can confirm that Tailwind CSS styles are working for the not-found pages 👍
